scripts/pylib/twister/twisterlib/testplan.py

```diff
@@ -562,7 +562,7 @@ def load_from_file(self, file, filter_platform=[]):
             instance_list = []
             for ts in jtp.get("testsuites", []):
                 logger.debug(f"loading {ts['name']}...")
-                testsuite = ts["name"]
+                testsuite = os.path.normpath(ts["name"])
```

Comment:
It will work when you generate plan on Linux and load it on Windows:
```
>>> os.path.normpath('twister-out/twister.log')
'twister-out\\twister.log'
```
When you generate plan on Windows and try to load it on Linux, it will not work:
```
>>> os.path.normpath('twister-out\\twister.log')
'twister-out\\twister.log'
```

IMO testcase name should be also changed in `TestSuite` class (e.g. to use only `/` separator).
Or, if we do not need to support that case (generate test plan on Windows and load on Linux), then this change will be enough.

Reply:
strange, why I get different result, see my log

`ubuntu@ubuntu-OptiPlex-7050:/home/shared/disk/zephyr_project/zephyr_test/zephyr$ ipython3
Python 3.10.12 (main, Jun 11 2023, 05:26:28) [GCC 11.4.0]
Type 'copyright', 'credits' or 'license' for more information
IPython 7.31.1 -- An enhanced Interactive Python. Type '?' for help.

In [1]: import os

In [2]: print(os.path.normpath('twister-out\\twister.log'))
twister-out\twister.log

`

Reply:
My first log was from Windows, second from Linux. When you use path with backslashes (generated in Windows) and test it on Linux, then you still have backslashes in the result.

Reply:
@gchwier , sorry there is  a paste error in my log, as you can see the `\\` is convert to `\`

Reply:
@gchwier thank a lot to point this scenario out, I add a check as below to specially handle it
`
                if testsuite not in self.testsuites:
                # this could be a path issue in testplan and run envrionment
                # try convert it
                # so far we know window plan to linux envirnmont has such problem
                    if "\\" in testsuite:
                        testsuite = testsuite.replace("\\", "/")
                    else:
                        logger.warning(f"{testsuite} is not supported")
                        continue
`